### PR TITLE
GIX-1543: Allow user participate if error getting country

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

If there is an error getting the country of the user, we enable the button to participate.

# Changes

No functionality changes.

# Tests

* Add a test in the ParticipateButton and ProjectDetail with the scenario that getting the country fails.

### CODE REVIEW

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts [hunk 1]
```diff
@@ -349,13 +352,7 @@ sale_buyer_count ${saleBuyerCount} 1677707139456
         });
       });
 
-      it("should participate with user interaction if there is no open ticket.", async () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/ProjectDetail.spec.ts:352]: I moved the setup of this test to a before each inside a describe to reuse it.

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts [hunk 2]
```diff
@@ -368,47 +365,128 @@ sale_buyer_count ${saleBuyerCount} 1677707139456
...
+              myCommitment: undefined,
+            } as SnsSwapCommitment)
+            .mockResolvedValue({
+              rootCanisterId: Principal.fromText(rootCanisterId),
+              myCommitment: finalCommitment,
+            } as SnsSwapCommitment);
         });
 
-        const { container } = render(ProjectDetail, props);
+        it("when no restricted countries", async () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/ProjectDetail.spec.ts:399]: This is the previous test.

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts [hunk 3]
```diff
@@ -368,47 +365,90 @@ sale_buyer_count ${saleBuyerCount} 1677707139456
...
-        );
+          expect(await projectDetail.hasCommitmentAmount()).toBe(false);
+          await projectDetail.participate({ amount: amountICP });
+          expect(await projectDetail.getCommitmentAmount()).toBe(
+            formattedAmountICP
+          );
+        };
+
+        it("when no restricted countries", async () => {
+          await participateInSwap();
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/ProjectDetail.spec.ts:422]: This is the same test as it was before.

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts [hunk 4]
```diff
@@ -368,47 +365,90 @@ sale_buyer_count ${saleBuyerCount} 1677707139456
...
+          );
 
-        await waitFor(async () =>
-          expect(await projectDetail.getStatus()).toBe(
-            "Accepting Participation"
-          )
-        );
+          await waitFor(async () =>
+            expect(
+              await projectDetail.getParticipateButton().isDisabled()
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/ProjectDetail.spec.ts:410]: This is the only change I did to the previous test. I thought it was more accurate to wait for the button to be enabled than to wait for the status.

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts [hunk 5]
```diff
@@ -368,47 +365,128 @@ sale_buyer_count ${saleBuyerCount} 1677707139456
...
-          formattedAmountICP
-        );
+          expect(await projectDetail.hasCommitmentAmount()).toBe(false);
+          await projectDetail.participate({ amount: amountICP });
+          expect(await projectDetail.getCommitmentAmount()).toBe(
+            formattedAmountICP
+          );
+        });
+
+        it("when restricted countries and user is from another country", async () => {
```
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/tests/lib/pages/ProjectDetail.spec.ts:425]: Do I see correctly that this file does not have a test where the button is disabled because the user is in a restricted country?
Will you add it in a different PR?
Or why do we need a test in this file where the button is enabled but not one where it is disabled?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/pages/ProjectDetail.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/ProjectDetail.spec.ts:425]: That's a good question. I wasn't sure what should and should not be here.

In the end, I decided to add at this level only the integration tests of the whole participation flow in different scenarios.

The specific edge cases where participating is not enabled are mostly in the ParticipateButton.

But I can add the test for the restricted countries in another PR as well.

[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/ProjectDetail.spec.ts:425]: it is asymetric. I was even thinking on adding this integrations tests in another file precisely because of that.
